src/cpp/Robot.cpp

```diff
@@ -10,13 +10,13 @@
 #include <Commands/Scheduler.h>
 #include <SmartDashboard/SmartDashboard.h>
 
-OI Robot::oi;
-DriveBase Robot::driveBase;
-Gripper Robot::gripper;
-Wrist Robot::wrist;
-Elevator Robot::elevator;
-
 void Robot::RobotInit() {
+	Robot::oi = new OI();
```

Comment:
The comment about never using the `new` keyword in C++ anymore aside, this code doesn't compile. You need the lines you deleted above to provide definitions of the static variables. At the moment, you've only specified declarations, and the linker is throwing errors about undefined references.

If you really don't want the static definitions, provide functions like the following instead.
```
OI& Robot::OI() {
  static OI oi;
  return oi;
}
```
This is known as the construct-on-first-use idiom. It's how people generally work around the non-deterministic static construction order problem.